packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2464,6 +2489,397 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+  ///   character found.
+  final String label;
+
+  /// Returns the [label] with any accelerator markers removed.
+  ///
+  /// This getter just calls [stripAcceleratorMarkers] with the [label].
+  String get displayLabel => stripAcceleratorMarkers(label);
+
+  /// The optional [MenuAcceleratorChildBuilder] which is used to build the
+  /// widget that displays the label itself.
+  ///
+  /// The [defaultLabelBuilder] function serves as the default value for
+  /// [builder], rendering the label as a [RichText] widget with appropriate
+  /// [TextSpan]s for rendering the label with an underscore under the selected
+  /// accelerator for the label when accelerators have been activated.
+  ///
+  /// {@macro flutter.material.menu_anchor.menu_accelerator_child_builder.args}
+  final MenuAcceleratorChildBuilder builder;
+
+  /// Whether [label] contains an accelerator definition.
+  ///
+  /// {@macro flutter.material.menu_anchor.menu_accelerator_label.label}
+  bool get hasAccelerator => RegExp(r'&(?!&)').hasMatch(label);
+
+  /// Serves as the default value for [builder], rendering the label as a
+  /// [RichText] widget with appropriate [TextSpan]s for rendering the label
+  /// with an underscore under the selected accelerator for the label when the
+  /// [index] is non-negative, and a [Text] widget when the [index] is negative.
+  ///
+  /// {@macro flutter.material.menu_anchor.menu_accelerator_child_builder.args}
+  static Widget defaultLabelBuilder(
+    BuildContext context,
+    String label,
+    int index,
+  ) {
+    if (index < 0) {
+      return Text(label);
+    }
+    final TextStyle defaultStyle = DefaultTextStyle.of(context).style;
+    return RichText(
+      text: TextSpan(
+        children: <TextSpan>[
+          if (index > 0) TextSpan(text: label.substring(0, index), style: defaultStyle),
+          TextSpan(
+              text: label.substring(index, index + 1),
+              style: defaultStyle.copyWith(decoration: TextDecoration.underline)),
+          if (index < label.length - 1) TextSpan(text: label.substring(index + 1), style: defaultStyle),
+        ],
+      ),
+    );
+  }
+
+  /// Strips out any accelerator markers from the given [label], and unescapes
+  /// any escaped ampersands.
+  ///
+  /// If [setIndex] is supplied, it will be called before this function returns
+  /// with the index in the returned string of the accelerator character.
+  ///
+  /// {@macro flutter.material.menu_anchor.menu_accelerator_label.label}
+  static String stripAcceleratorMarkers(String label, {void Function(int index)? setIndex}) {
+    int quotedAmpersands = 0;
+    final StringBuffer displayLabel = StringBuffer();
+    int acceleratorIndex = -1;
+    for (int i = 0; i < label.length; i += 1) {
+      // Stop looking one before the end, since a single ampersand at the end is
+      // just treated as a quoted ampersand.
+      if (label[i] != '&' || i == label.length - 1) {
```

Comment:
Not sure how realistic this is but... hopefully people won't be trying to use 𨨏 as accelerator character as this would yield half of the surrogate pair :-).

Reply:
Good point.  I'll convert to using `Character`s for the search, since I don't want part of a surrogate pair triggering it. Non-ASCII accelerators are uncommon, though (as far as I understand).

Reply:
Or perhaps this can be done using String.runes so as not to create dependency on Characters?
I can see people use non-Latin (non-ASCII) characters but I guess surrogate pairs are less likely.

Reply:
Flutter already has a dependency on Characters.

Reply:
+1